sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Authentication/ContainerRegistryChallengeAuthenticationPolicy.cs

```diff
@@ -34,66 +34,64 @@ namespace Azure.Containers.ContainerRegistry
     internal class ContainerRegistryChallengeAuthenticationPolicy : BearerTokenChallengeAuthenticationPolicy
     {
         private readonly IContainerRegistryAuthenticationClient _authenticationClient;
+        private readonly ContainerRegistryRefreshTokenCache _refreshTokenCache;
 
         public ContainerRegistryChallengeAuthenticationPolicy(TokenCredential credential, string aadScope, IContainerRegistryAuthenticationClient authenticationClient)
+            : this(credential, aadScope, authenticationClient, null, null, null)
+        {
+        }
+
+        internal ContainerRegistryChallengeAuthenticationPolicy(TokenCredential credential, string aadScope, IContainerRegistryAuthenticationClient authenticationClient, TimeSpan? tokenRefreshOffset = null, TimeSpan? tokenRefreshRetryDelay = null, TimeSpan? tokenExpiryOffset = null)
             : base(credential, aadScope)
         {
             Argument.AssertNotNull(credential, nameof(credential));
             Argument.AssertNotNull(aadScope, nameof(aadScope));
 
             _authenticationClient = authenticationClient;
+            _refreshTokenCache = new ContainerRegistryRefreshTokenCache(credential, authenticationClient, tokenRefreshOffset, tokenRefreshRetryDelay, tokenExpiryOffset);
+        }
+
+        protected override Task AuthorizeRequestAsync(HttpMessage message, bool async)
```

Comment:
Is this called as part of Step 1? Might be worth adding some comments.

Reply:
Will do!